src/main/java/fi/aalto/cs/apluscourses/model/Course.java

```diff
@@ -231,21 +190,20 @@ private static String getCourseName(@NotNull JSONObject jsonObject, @NotNull Str
         String pluginName = requiredPluginsJson.getString(pluginId);
         requiredPlugins.put(pluginId, pluginName);
       } catch (JSONException ex) {
-        throw new MalformedCourseConfigurationFileException(path,
+        throw new MalformedCourseConfigurationFileException(source,
             "Expected id-name-pairs in requiredPlugins object", ex);
       }
     }
     return requiredPlugins;
   }
 
-
   @NotNull
   @Override
-  public Module getModule(@NotNull String moduleName) throws NoSuchModuleException {
+  public Optional<Module> getModuleOptional(@NotNull String moduleName) {
```

Comment:
Not sure what is an advantage of using `Optional` instead of plain nullable reference, but it seems to be customary in Java so I guess that's OK. :)

Reply:
Just my instinct to avoid nulls :smile: But since we're consistently using the `@Nullable` and `@NotNull` annotations, I think a nullable reference would be completely fine here as well